feat: Add featured listing styles

laurelfulford opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR contains a first swing at the featured listing styles for the Newspack theme. 

I think it made sense to tackle them in the theme rather than the plugin because the styles might differ from theme to theme, and that the styles are applied to blocks in two different plugins (Newspack Blocks, Newspack Listings), on top of the archive and search styles in the theme. Happy to discuss that further if moving them makes sense though!

See https://github.com/Automattic/newspack-listings/issues/131.

### How to test the changes in this Pull Request:

1. Enable Featured Listings on your test site by adding `define( 'NEWSPACK_LISTINGS_FEATURED_ENABLED', true );` to your wp-config.php file. 
2. Create a number of listings and mark some as 'Featured'.
3. Add the Homepage Posts block, Post Carousel Block and Curated List Listings block; make sure at least one Featured listing appears in all three. 
4. Navigate to Customizer > Colors and change your primary colour. 
5. View your post on the front-end and in the editor; confirm that your featured listings have a little "ribbon" icon that uses the site's primary colour:

Homepage Posts block: 

![image](https://user-images.githubusercontent.com/177561/178581274-73e4c9c5-87e6-4b18-bac2-fc22f74e54e2.png)

Post Carousel:

![image](https://user-images.githubusercontent.com/177561/178581206-aa75b5e2-d4bb-4f5a-a730-c78068abdcd9.png)

Curated List (note: the featured image classes are not yet added to the curated list block in the editor; issue filed [here](https://github.com/Automattic/newspack-listings/issues/282)). 

![image](https://user-images.githubusercontent.com/177561/178581154-8a68252d-c402-4334-b3fe-852957e1b51a.png)

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo reviewed (changes requested):
This is 99% there, but I noticed a tiny gap in the "ribbon" icon when I place a Homepage Posts in the sidebar widget and enable listing post types:

<img width="376" alt="Screen Shot 2022-08-01 at 4 47 08 PM" src="https://user-images.githubusercontent.com/2230142/182258707-36ba1789-cf3b-4429-a1ec-86ffdbf8ba27.png">

laurelfulford commented:
Oh, good catch @dkoo! I used `em` to keep the ribbon proportionate to the text, but it looks like that resulted in a bit of a pixel rounding issue at some font sizes (I was able to recreate this in the editor, too, by scaling up the font size of the block!). 

I've added a pixel of wiggle room that seems to help! I also switched how the bottom "cut in" was measured -- it was originally a pixel value but got a little disproportionate at smaller font sizes. It now uses `em`, too 🙂 

Just let me know if anything else seems funky to you!

dkoo reviewed (approved):
Looking good now!

<img width="244" alt="Screen Shot 2022-08-05 at 4 13 08 PM" src="https://user-images.githubusercontent.com/2230142/183218210-5f5b7cce-96a5-4527-88fb-65f57179fe41.png">
